crypto/threads_win.c

```diff
@@ -569,6 +576,44 @@ int CRYPTO_atomic_add(int *val, int amount, int *ret, CRYPTO_RWLOCK *lock)
     return 1;
 }
 
+int CRYPTO_atomic_add64(uint64_t *val, uint64_t op, uint64_t *ret,
```

Comment:
you need to run make update to get this in the exported ordinals list, its why CI is failing here

Reply:
Thanks! I was wondering what this error means.
That ancient piece of history Windows 2003 is using `nmake` to compile, so I ran the command on macOS instead. I see the two new functions were added to `util/libcrypto.num`. I hope that's good enough.

Reply:
Not quite.  The CI test for symbol presence is failing, need to look at why.

also, if you're going to add new exported library symbols, the docs check is going to fail if you don't document them in the appropriate pod file

Reply:
`doc/man3/CRYPTO_THREAD_run_once.pod` contains all related function, so I added the description there.

Reply:
Building on Windows works only with `nmake`. I do have `GNU Make 4.4.1`, but both fail to perform the `update` target. So my options are macOS 14.4.1 Intel, and Ubuntu 22.04.4 LTS.

```
nmake update
Microsoft (R) Program Maintenance Utility Version 10.00.40219.01
Copyright (C) Microsoft Corporation.  All rights reserved.

NMAKE : fatal error U1073: don't know how to make 'update'
Stop.

make update
makefile:78: *** missing separator.  Stop.
```

I also noticed that apart from `threads_win.c`, the atomic functions are also defined in `threads_pthread.c` and `threads_none.c`. Do I need to implement them there as well?

Reply:
Or maybe I need to add them to `./test/threadstest.c`?

Reply:
you do need to add the definitions to threads_pthread.c and threads_none.c, as only one of the three (win/pthreads/none) gets compiled for the build and you need a definition for the exported symbols

As for why make update isn't working.  I expect its got something to do with the version of nmake on vs2010.  Its worked for me on VS2022 community edition.  note VS2010 isn't an officially [supported platform](https://www.openssl.org/policies/general-supplemental/platforms.html)

I'd focus on getting the implementations defined on other platforms first, then worry about being able to make the update script.

Reply:
Added definitions to `threads_pthread.c` and `threads_none.c`. Added tests to `threadstest.c`.